bootloader/grub.go

```diff
@@ -612,3 +636,15 @@ func (g *grub) BootChain(runBl Bootloader, kernelPath string) ([]BootFile, error
 
 	return chain, nil
 }
+
+// ConstructShimEfiLoadOption returns a serialized load option for the shim
+// binary. It should be called on a UefiBootloader.
+func (g *grub) EfiLoadOptionParameters() (description string, assetPath string, optionalData []byte, err error) {
+	assetPath, err = g.getGrubShimBinaryFullPath()
+	if err != nil {
+		return "", "", nil, err
+	}
+	description = "ubuntu"
+	optionalData = make([]byte, 0)
+	return description, assetPath, optionalData, nil
```

Comment:
I think it would make sense to be able to return nil here instead of an empty slice

Reply:
Yes, I agree. I've added a test to make sure `optionalData = nil` produces an equivalent load option to `optionalData = make([]byte, 0)` -- and they are. I changed this grub code to use `nil` instead.